### ISSUE TITLE
Windows 10: recognize several common dialog classes from XAML and other controls

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#8405]
Hi,

Related to #8339:

There are dialogs powered by XAML that are not recognized by NVDA as proper dialogs. These include app removal dialog in Settings/Apps/App and Features, reset dialog in Settings/Update and Security/Recovery/Reset, and potentially others. For now, NVDA does not read them natively, and in order to announce them, one needs to install Windows 10 App Essentials add-on.

Technical: in the old days, automation ID's weren't useful, but that's slowly changing. Thankfully these controls have unique UIA class names, including Shell_SystemDialog, Popup and others. Checking for class names could serve as one of several tests to detect dialogs.

Thanks.

**Comment by josephsl** [comment]
Hi,

This and #8339 will be combined into a single PR. Thanks.

**Comment by josephsl** [comment]
Hi,

Actually, they will be a separate PR just for this issue so it can go live as part of 2018.3 at the earliest. Thanks.
